### PR TITLE
Fix plasma mixer power consumption and overflow

### PR DESCRIPTION
**Pull request by HoleFish**
supersedes https://github.com/GTNewHorizons/GT5-Unofficial/pull/2777

### CODE REVIEW

**Comment by Dream-Master** [comment]
@Nxer

**Review by Nxer — APPROVED** [review]

**Comment by Nxer** [comment]
Wait the text part. Looks ok.

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_TranscendentPlasmaMixer.java [hunk 1]
```diff
@@ -172,24 +172,30 @@ public RecipeMap<?> getRecipeMap() {
...
+            BigInteger recipeEU;
+
             @NotNull
             @Override
             protected CheckRecipeResult validateRecipe(@Nonnull GT_Recipe recipe) {
-                mWirelessEUt = 10L * (long) recipe.mEUt * (long) multiplier;
-                if (getUserEU(ownerUUID).compareTo(BigInteger.valueOf(mWirelessEUt * recipe.mDuration)) < 0) {
-                    return CheckRecipeResultRegistry.insufficientPower(mWirelessEUt * recipe.mDuration);
+                BigInteger availableEU = getUserEU(ownerUUID);
+                recipeEU = BigInteger.valueOf(10L * recipe.mEUt * recipe.mDuration);
```
Comment (OneEyeMaker) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_TranscendentPlasmaMixer.java:181]: Can this calculation cause `long` overflow? If so, shouldn't we use `BigInteger::multiply' instead?

Comment (Nxer) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_TranscendentPlasmaMixer.java:181]: > Can this calculation cause `long` overflow? If so, shouldn't we use `BigInteger::multiply' instead?

No, overflow happens when parallel is huge enough.

**Review by Nxer — APPROVED** [review]
Able to merge.

Extra be careful not to cause long type overflow when writing new recipes.

**Review by OneEyeMaker — APPROVED** [review]

**Pull request merged**